### ISSUE TITLE
Add support for ANOVA-like analyses in multiGSEA()

### ISSUE THREAD

**Issue by lianos** [lianos/multiGSEA#11]
Currently we only support testing a specific contrast (either via a coefficient or a contrast vector), but we often do differential expression analysis where we want to test mean shifts in several groups of samples, which we would like to support here.

For instance, if we had a 3-group experiment, ie. a `group` factor had `"control"`, `"treatment01"`, and `"treatment02"` levels, we might do a limma analysis like so:

```r
des <- model.matrix(~ group)
fit <- eBayes(lmFit(y, des))
res <- topTable(fit, coef = 2:3, ...)
```

We want to support that in a GSEA-like manner.

**Comment by dktanwar** [comment]
Hi @lianos 

Is there any update on this?

**Comment by lianos** [comment]
Yes, there is some basic support for ANVOA analyses followed up with enrichment testing.

I'm only supporting GSEA testing using the `"enrichtest"` method, which is a wrapper to `limma::kegga` (which, itself is like goseq).

You can see an example of how that works in [this unit test](https://github.com/lianos/multiGSEA/blob/5ec7710d6ce3146a0503628a64f5c97d4cd7ada1/tests/testthat/test-enrichtest.R#L148).

I just added that test along with some bugfixes to support that in the latest `@develop` branch, so you will have to `devtools::install_github("lianos/multiGSEA@develop")` to get it to work.

Let me know if you have any issues.
